### ISSUE TITLE
fanc.pairs.FourDNucleomePairGenerator seems to freeze?

### ISSUE THREAD

**Issue by gabedolpten** [vaquerizaslab/fanc#81]
Hi, I am trying to load a FourDNucleomePairGenerator file with Fan-C, using:

tb = fanc.pairs.FourDNucleomePairGenerator(path_to_pairs)

where path_to_pairs leads to an indexed .pairs.gz file. However, this command seems to freeze and does not resolve. Do you know why this is happening?

**Comment by gabedolpten** [comment]
PS I am on 0.9.21

**Comment by kaukrise** [comment]
I think this is just a stupid mistake I made. Can you try the following version?

[fanc-0.9.22b0.tar.gz](https://github.com/vaquerizaslab/fanc/files/7197088/fanc-0.9.22b0.tar.gz)

(let me know if you need help installing from tarball)

**Comment by gabedolpten** [comment]
Oh, ok, got it. This works well! Just wondering, is it possible to query 4DNucPairGenerator as with a fanc .pairs file, ie with ––

querystr='{}:{}-{}'.format(chrom, start-100, end+100)
tb.pairs((querystr, querystr))

**Comment by kaukrise** [comment]
No, that won't work, you'll have to generate a pairs file using the generator as input first. The pairs file can be queried that way.

**Comment by kaukrise** [comment]
I would recommend using the FAN-C command line tool for this, though - it is much more straightforward compared to the API. Once you have the pairs object you can easily continue in the shell
